client/my-sites/themes/thanks-modal.jsx

```diff
@@ -334,19 +330,6 @@ const ConnectedThanksModal = connect(
 		// Note: Gutenberg buttons will only show if the homepage is a page.
 		const shouldEditHomepageWithGutenberg = shouldCustomizeHomepageWithGutenberg( state, siteId );
 
-		const isAtomic = isSiteAtomic( state, siteId );
-		const isJetpack = isJetpackSite( state, siteId );
-		const hasAutoLoadingHomepage = themeHasAutoLoadingHomepage( state, currentThemeId, siteId );
-
-		// Atomic & Jetpack do not have auto-loading-homepage behavior, so we trigger the layout picker for them.
-		const customizeUrl =
-			( isAtomic || isJetpack ) && hasAutoLoadingHomepage
-				? addQueryArgs(
-						{ 'new-homepage': true },
```

Comment:
I'm removing this part because the param `?new-homepage=true` is meant to open the modal in the Post editor to choose a starter page template for a new homepage, assuming Atomic and Jetpack sites don't support auto-loading-homepage behavior. However, Atomic sites support it. See https://github.com/Automattic/wp-calypso/pull/65976. 

In this PR, we always pass `dont_change_homepage: true`, which skips the auto-loading-homepage behavior and that works on Atomic & Jetpack. See https://github.com/Automattic/wp-calypso/pull/65846#issuecomment-1192650587 and fbhepr%2Skers%2Sjcpbzfu%2Ssrngher%2Qcyhtvaf%2Snhgbybnq%2Qubzrcntr%2Qercynprzrag.cuc%3Se%3Q923495oq%2320-og 

I'll test these scenarios again to confirm we are safely removing that param.